### PR TITLE
Optional requirements

### PR DESCRIPTION
**Pull request by patriciachrysy**
# 👋 Micronaut Reviewer 

[To Do list: list structure](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m1_list_structure.md)

Today, i'm working on the project third step :

### Project requirements
- [x] Remove all hardcoded items from the tasks array.
- [x] Create a new JavaScript file for the new functionality.
- [x] Implement a function for adding a new task (add a new element to the array).
- [x] Implement a function for editing task descriptions.
- [x] Implement a function for deleting a task (remove an element from the array).
- [x] Implement a function for the "Clear all completed" button (use `filter()` method).
- [x] By default new tasks should have the property `completed` set to `false` and the property `index` set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- [x] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
- [x] All changes to the To Do List should be saved in local storage.

I also did the optional requirement of the previous step:
## Additional requirements

- [x] Add a new JavaScript file and import it as a module:    
    - [x] it will contain methods related to drag / drop and sorting functionality.
- [x] Add event listeners to list items (`dragstart`, `dragover`, `drop`).
- [x] Update items object's values for `index` key upon user actions.

Happy Review :)

### CODE REVIEW

File: src/taskActionsHelper.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  return taskList;
+};
+
+export const deleteTask = (taskIndex, taskList) => {
+  taskList = taskList.filter((task) => task.index !== parseInt(taskIndex, 10));
+
+  taskList.map((task, pos) => {
+    task.index = pos;
+    return 0;
```
Comment (Davidosky007) [inline src/taskActionsHelper.js:37]: * Kindly add the delete functionality to your UI( user interface) as currently, i can't delete a task [Check project requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Implement%20a%20function%20for%20deleting%20a%20task%20(remove%20an%20element%20from%20the%20array).)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @patriciachrysy 👋, 

Good job so far! 💪 
* Well written PR ✔️ 
* No linter errors ✔️ 
* Descriptive commit messages ✔️ 
* Well done implementing the drag and drop functionality ✔️ 
* Nice looking readme file ✔️ 
## There are some issues that you still need to work on to go to the next project but you are almost there!
you-can-do-this-gif


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/taskActionsHelper.js (hunk 1 above)
Comment (patriciachrysy) [reply src/taskActionsHelper.js:37]: Hello Reviewer, thanky for your review, the deleteTask functionnality is added, you need to click on a task description in order to see the 'bin' icon that will delete the task when you click on it. I followed the requirements, kindly take time to check again. Thank you.

**Comment by patriciachrysy** [comment]
Hello Reviewer, thanky for your review, the deleteTask functionnality is added, you need to click on a task description in order to see the 'bin' icon that will delete the task when you click on it. I followed the requirements, kindly take time to check again. Thank you.

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @patriciachrysy , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

**Pull request merged**